mlflow/utils/autologging/logging_and_warnings.py

```diff
@@ -0,0 +1,287 @@
+import os
```

Comment:
Introduce utilities for rerouting and silencing MLflow & non-MLflow warnings, providing support for multithreaded contexts to ensure that autologging warnings are not leaked.